dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileBig.cpp

```diff
@@ -94,6 +99,59 @@ ColumnFilePersistedPtr ColumnFileBig::deserializeMetadata(const DMContext & cont
     return std::shared_ptr<ColumnFileBig>(dp_file);
 }
 
+ColumnFilePersistedPtr ColumnFileBig::createFromCheckpoint(DMContext & context, //
+                                                           const RowKeyRange & target_range,
+                                                           ReadBuffer & buf,
+                                                           UniversalPageStoragePtr temp_ps,
+                                                           TableID ns_id,
+                                                           WriteBatches & wbs)
+{
+    UInt64 file_page_id;
+    size_t valid_rows, valid_bytes;
+
+    readIntBinary(file_page_id, buf);
+    readIntBinary(valid_rows, buf);
+    readIntBinary(valid_bytes, buf);
+
+    // get target dtfile s3 key
+    auto remote_page_id = UniversalPageIdFormat::toFullPageId(UniversalPageIdFormat::toFullPrefix(StorageType::Data, ns_id), file_page_id);
+    auto remote_data_location = temp_ps->getCheckpointLocation(remote_page_id);
+    auto remote_file_id = temp_ps->getNormalPageId(remote_page_id);
+    auto file_id = UniversalPageIdFormat::getU64ID(remote_file_id);
+
+    const auto & lock_key_view = S3::S3FilenameView::fromKey(*(remote_data_location->data_file_id));
+    S3::DMFileOID file_oid{
+        .store_id = lock_key_view.store_id,
+        .table_id = ns_id,
+        .file_id = file_id};
+    RUNTIME_CHECK(lock_key_view.asDataFile().toFullKey() == S3::S3Filename::fromDMFileOID(file_oid).toFullKey());
+
+    auto data_key = S3::S3Filename::fromDMFileOID(file_oid).toFullKey();
+    auto delegator = context.path_pool->getStableDiskDelegator();
+    auto & storage_pool = context.storage_pool;
+    auto new_local_file_id = storage_pool->newDataPageIdForDTFile(delegator, __PRETTY_FUNCTION__);
+    PS::V3::CheckpointLocation loc{
+        .data_file_id = std::make_shared<String>(data_key),
+        .offset_in_file = 0,
+        .size_in_file = 0,
+    };
+    wbs.data.putRemoteExternal(new_local_file_id, loc);
+
+    // create a local file with only needed metadata
+    auto parent_path = delegator.choosePath();
+    auto new_dmfile = DMFile::create(new_local_file_id, parent_path);
```

Comment:
Why do we need to create a read dtfile on local disk?

Reply:
Should not need this step after this PR merged: https://github.com/pingcap/tiflash/pull/7047

Reply:
Yes, and it will also speed up FAP.